integrationtests/all-embedded-query-it/src/test/java/org/infinispan/all/embeddedquery/LocalCacheTest.java

```diff
@@ -0,0 +1,573 @@
+      loadTestingData();
+      CacheQuery cacheQuery = createCacheQuery(cache, "blurb", "playing");
+
+      List<Object> found = cacheQuery.list();
+
+      assertEquals(1, found.size());
+      assertTrue(found.get(0).equals(person1));
+
+      person1.setBlurb("Likes pizza");
+      cache.put(key1, person1);
+
+      cacheQuery = createCacheQuery(cache, "blurb", "pizza");
+
+      found = cacheQuery.list();
+
+      assertEquals(1, found.size());
+      assertTrue(found.get(0).equals(person1));
+      StaticTestingErrorHandler.assertAllGood(cache);
+   }
+
+   @Test
+   public void testAdded() {
+      assertIndexingKnows(cache);
+      loadTestingData();
+      assertIndexingKnows(cache, Person.class);
+
+      CacheQuery cacheQuery = createCacheQuery(cache, "name", "Goat");
+      List<Object> found = cacheQuery.list();
+
+      assertEquals("Size of list should be 2", 2, found.size());
+      assertTrue(found.contains(person2));
+      assertTrue(found.contains(person3));
+      assertTrue("This should not contain object person4", !found.contains(person4));
+
+      person4 = new Person();
+      person4.setName("Mighty Goat");
+      person4.setBlurb("Also eats grass");
+
+      cache.put("mighty", person4);
+
+      cacheQuery = createCacheQuery(cache, "name", "Goat");
+      found = cacheQuery.list();
+
+      assertEquals("Size of list should be 3", 3, found.size());
+      assertTrue(found.contains(person2));
+      assertTrue(found.contains(person3));
+      assertTrue("This should now contain object person4", found.contains(person4));
+      StaticTestingErrorHandler.assertAllGood(cache);
+   }
+
+   @Test
+   public void testRemoved() {
+      loadTestingData();
+
+      CacheQuery cacheQuery = createCacheQuery(cache, "name", "Goat");
+      List<Object> found = cacheQuery.list();
+
+      assertEquals(2, found.size());
+      assertTrue(found.contains(person2));
+      assertTrue("This should still contain object person3", found.contains(person3));
+
+      cache.remove(key3);
+
+      cacheQuery = createCacheQuery(cache, "name", "Goat");
+      found = cacheQuery.list();
+
+      assertEquals(1, found.size());
+      assertTrue(found.contains(person2));
+      assertTrue("The search should not return person3", !found.contains(person3));
+      StaticTestingErrorHandler.assertAllGood(cache);
+   }
+
+   @Test
+   public void testUpdated() {
+      loadTestingData();
+
+      CacheQuery cacheQuery = createCacheQuery(cache, "name", "Goat");
+      List<Object> found = cacheQuery.list();
+
+      assertEquals("Size of list should be 2", 2, found.size());
+      assertTrue("The search should have person2", found.contains(person2));
+
+      cache.put(key2, person1);
+
+      cacheQuery = createCacheQuery(cache, "name", "Goat");
+      found = cacheQuery.list();
+      
+      assertEquals("Size of list should be 1", 1, found.size());
+      assertTrue("Person 2 should not be found now", !found.contains(person2));
```

Comment:
Just an observation - since we are playing a lot with persons in this cache - maybe it is worth to create a custom assertion for those? Here is an example <http://www.pholser.com/writings/customassertions.html>